modules/experimental/src/gpgpu/point-in-polygon/gpu-point-in-polygon.js

```diff
@@ -0,0 +1,183 @@
+    this.indexBuffer = new Buffer(gl, {
+      target: GL.ELEMENT_ARRAY_BUFFER,
+      accessor: {type: GL.UNSIGNED_SHORT}
+    });
+
+    // transform to generate polygon texture
+    this.polyTextureTransform = new Transform(gl, {
+      id: `polygon-texture-creation-transform`,
+      elementCount: 0,
+      _targetTexture: this.polygonTexture,
+      _targetTextureVarying: 'v_polygonColor',
+      vs: POLY_TEX_VS,
+      drawMode: GL.TRIANGLES,
+      isIndexed: true,
+      sourceBuffers: {
+        a_position: this.positionBuffer,
+        a_polygonID: this.idBuffer,
+        indices: this.indexBuffer
+      }
+    });
+
+    // transform to perform filtering
+    this.filterTransform = new Transform(gl, {
+      id: 'filter transform',
+      vs: FILTER_VS,
+      modules: [textureFilterModule],
+      varyings: ['filterValueIndex']
+    });
+  }
+
+  _updateResources(vertices, indices, ids, vertexCount) {
+    const boundingBox = getBoundingBox(vertices, vertexCount);
+    const [xMin, yMin, xMax, yMax] = boundingBox;
+    const width = xMax - xMin;
+    const height = yMax - yMin;
+    const whRatio = width / height;
+    const {textureSize} = this;
+
+    // calculate max texture size with same aspect ratio
+    let texWidth = textureSize;
+    let texHeight = textureSize;
+    if (whRatio > 1) {
+      texHeight = texWidth / whRatio;
+    } else {
+      texWidth = texHeight * whRatio;
+    }
+
+    this.boundingBox = boundingBox;
+    this.polygonTexture.resize({width: texWidth, height: texHeight, mipmaps: false});
+    this.positionBuffer.setData(new Float32Array(vertices));
+    this.idBuffer.setData(new Float32Array(ids));
+    this.indexBuffer.setData(new Uint16Array(indices));
+    this.polyTextureTransform.update({
+      elementCount: indices.length,
+      _targetTexture: this.polygonTexture
+    });
+
+    this.polyTextureTransform.run({
+      uniforms: {
+        boundingBoxOriginSize: [xMin, yMin, width, height]
+      }
+    });
+  }
+}
+
+// Helper methods
+
+function getBoundingBox(positions, vertexCount) {
+  let yMin = Infinity;
+  let yMax = -Infinity;
+  let xMin = Infinity;
+  let xMax = -Infinity;
+  let y;
+  let x;
+
+  for (let i = 0; i < vertexCount; i++) {
+    x = positions[i * 2];
+    y = positions[i * 2 + 1];
+    yMin = y < yMin ? y : yMin;
+    yMax = y > yMax ? y : yMax;
+    xMin = x < xMin ? x : xMin;
+    xMax = x > xMax ? x : xMax;
+  }
+
+  return [xMin, yMin, xMax, yMax];
+}
+
+function triangulatePolygons(polygons) {
+  const SIZE = 2;
+  const vertices = [];
+  const indices = [];
+  const ids = [];
+  let count = 0;
+  let polygonId = 0;
+  for (let i = 0; i < polygons.length; i++) {
+    const normalized = Polygon.normalize(polygons[i], SIZE);
+    const curVertices = normalized.positions || normalized;
+    const curCount = curVertices.length / SIZE;
+    const curIds = new Float32Array(curCount).fill(polygonId);
+    vertices.push(...curVertices);
+    ids.push(...curIds);
```

Comment:
How big can these arrays get? Would recommend pre-allocating the arrays to avoid re-allocating and they dynamically extend.

Reply:
We can add array manager support in the future PRs, that can be independent of this feature and used across all luma modules.